src/bson_mapper/bson_streambuf.hpp

```diff
@@ -111,12 +111,12 @@ class BSON_MAPPER_API char_array_streambuf : public std::streambuf {
      * This reads and returns a character from the buffer, and increments the read pointer.
      * @return [description]
      */
-    virtual BSON_MAPPER_PRIVATE int underflow() override;
+    int underflow() final;
```

Comment:
Should these all be `final override`? That way they must override a virtual function in the base class.